### PR TITLE
🐛 Fixed permission issue allowing contributors to edit posts

### LINKED ISSUE #10238: Contributor should not be allowed to edit a post when not being a primary author
This issue is a follow up to this [fix](https://github.com/TryGhost/Ghost/pull/10214) and this [conversation](https://github.com/TryGhost/Ghost/commit/7c1840f0253be175df492396df5de0a815e689ca#r31539514). 

For now we only allow editing to **primary authors** for 'Contributors' and allow editing as a **co-author** for 'Authors'. More cleanup and test coverage is to come in https://github.com/TryGhost/Ghost/issues/10234

### PR DESCRIPTION
**Pull request by naz**
closes #10238

Changes as discussed in https://github.com/TryGhost/Ghost/commit/7c1840f0253be175df492396df5de0a815e689ca#r31539514

### CODE REVIEW

File: core/server/models/relations/authors.js [hunk 1]
```diff
@@ -314,22 +314,26 @@ module.exports.extendModel = function extendModel(Post, Posts, ghostBookshelf) {
...
+                return (context.user === postModel.related('authors').models[0].id);
+            }
+
+            function isCoAuthor() {
                 return postModel.related('authors').models.map(author => author.id).includes(context.user);
             }
 
             if (isContributor && isEdit) {
-                hasUserPermission = !isChanging('author_id') && !isChangingAuthors() && isCurrentOwner();
+                hasUserPermission = !isChanging('author_id') && !isChangingAuthors() && isPrimaryAuthor();
```
[the remark this answers is not in the document]
Comment (gargol) [reply core/server/models/relations/authors.js]: Idk tbh. From my logic seems like contributor always has fewer rights then author, but this case is not clear to me. But maybe contributor should be allowed to edit any post that she coauthored?

[the remark this answers is not in the document]
Comment (kirrg001) [reply core/server/models/relations/authors.js]: yeah i think so. as long as the contrib is 

- not able to change authors 
- is not able to publish

[the remark this answers is not in the document]
Comment (gargol) [reply core/server/models/relations/authors.js]: > not able to change authors

is checked in the same condition

> is not able to publish 

that stays as is and is handled in `permissible()` method in post model. I think changing it to `isCoAuthor` here should be fine :+1:

[the remark this answers is not in the document]
Comment (kirrg001) [reply core/server/models/relations/authors.js]: 👍

**Comment by kirrg001** [comment]
> Fixed permission issue allowing contributors to edit posts

Suggestion: Fixed contributors are able to delete draft posts as co author

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**